projects/ad7606x_fmc/common/ad7606x_bd.tcl

```diff
@@ -1,125 +1,217 @@
+      ad_ip_parameter adc_clk_generator CONFIG.CLKOUT1_REQUESTED_OUT_FREQ 100.000
+      ad_ip_parameter adc_clk_generator CONFIG.CLKOUT1_REQUESTED_PHASE 0.000
+      ad_ip_parameter adc_clk_generator CONFIG.CLKOUT1_REQUESTED_DUTY_CYCLE 50.000
+      ad_ip_parameter adc_clk_generator CONFIG.PRIM_SOURCE Global_buffer
+      ad_ip_parameter adc_clk_generator CONFIG.CLKIN1_UI_JITTER 0
+      ad_ip_parameter adc_clk_generator CONFIG.PRIM_IN_FREQ 100.000
+    
+      ad_connect  sys_cpu_clk adc_clk_generator/clk_in1
+      ad_connect  sys_cpu_resetn adc_clk_generator/resetn
+      ad_connect  adc_clk_generator/clk_out1 axi_ad7606x/external_clk
+    } else {
+      ad_connect  sys_cpu_clk axi_ad7606x/external_clk
+    }
+    
+    # interface connections
+    ad_connect  rx_db_o axi_ad7606x/rx_db_o
+    ad_connect  rx_db_i axi_ad7606x/rx_db_i
+    ad_connect  rx_db_t axi_ad7606x/rx_db_t
+    ad_connect  rx_rd_n axi_ad7606x/rx_rd_n
+    ad_connect  rx_wr_n axi_ad7606x/rx_wr_n
+    
+    ad_connect  rx_cs_n axi_ad7606x/rx_cs_n
+    ad_connect  rx_cnvst_n ad7606_pwm_gen/pwm_0
+    ad_connect  rx_busy axi_ad7606x/rx_busy
+    ad_connect  rx_first_data axi_ad7606x/first_data
+    
+    ad_connect  sys_cpu_clk axi_ad7606x_dma/s_axi_aclk
+    ad_connect  sys_cpu_clk ad7606_pwm_gen/s_axi_aclk
+    ad_connect  sys_cpu_resetn ad7606_pwm_gen/s_axi_aresetn
+    
+    ad_connect  axi_ad7606x/adc_clk ad7606x_adc_pack/clk
+    ad_connect  axi_ad7606x/adc_clk axi_ad7606x_dma/fifo_wr_clk
+    ad_connect  axi_ad7606x/adc_reset ad7606x_adc_pack/reset
+    ad_connect  axi_ad7606x/adc_valid ad7606x_adc_pack/fifo_wr_en
+    ad_connect  ad7606x_adc_pack/packed_fifo_wr axi_ad7606x_dma/fifo_wr
+    ad_connect  ad7606x_adc_pack/fifo_wr_overflow axi_ad7606x/adc_dovf
+    
+    for {set i 0} {$i < 8} {incr i} {
+      ad_connect axi_ad7606x/adc_data_$i ad7606x_adc_pack/fifo_wr_data_$i
+      ad_connect axi_ad7606x/adc_enable_$i ad7606x_adc_pack/enable_$i
+    }
+    
+    # interconnect
+    ad_cpu_interconnect  0x44a00000 axi_ad7606x
+    ad_cpu_interconnect  0x44a30000 axi_ad7606x_dma
+    ad_cpu_interconnect  0x44b00000 ad7606_pwm_gen
+    
+    # memory interconnect
+    ad_mem_hp1_interconnect sys_cpu_clk sys_ps7/S_AXI_HP1
+    ad_mem_hp1_interconnect sys_cpu_clk axi_ad7606x_dma/m_dest_axi
+    ad_connect sys_cpu_resetn axi_ad7606x_dma/m_dest_axi_aresetn
+    
+    # interrupt
+    ad_cpu_interrupt ps-13 mb-12 axi_ad7606x_dma/irq
+    }
+  1 {
+
+    # instantiation
+    create_bd_intf_port -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 ad7606_spi
+  
+    # spi engine
+    source $ad_hdl_dir/library/spi_engine/scripts/spi_engine.tcl
+    
+    set data_width    $ADC_TO_DMA_N_BITS
+    set async_spi_clk 1
+    set num_cs        1
+    set num_sdi       $NUM_OF_SDI
+    set sdi_delay     1
+    
+    set hier_spi_engine spi_ad7606
+    
+    spi_engine_create $hier_spi_engine $data_width $async_spi_clk $num_cs $num_sdi $sdi_delay
```

Comment:
this call is missing parameters. please check with:
https://github.com/analogdevicesinc/hdl/blob/main/library/spi_engine/scripts/spi_engine.tcl#L6

Reply:
Fixed.